server/aesdsocket.c

```diff
@@ -364,13 +276,274 @@ static void handle_socket()
+		status = listen(sockfd, 10);
+		if (status == -1)
+		{
+			syslog(LOG_ERR, "Listen failed:%d\n", status);
+			printf("Listen Failed\n");
+			exit(EXIT_FAILURE);
+		}
+		else
+		{
+			syslog(LOG_INFO, "Listen Successfull\n");
+			printf("Listen Successfull\n");
+		}
+
+		//Accept the socket connection
+		client_address_length = sizeof(struct sockaddr);
+
+		clifd = accept(sockfd, (struct sockaddr *) &client_address, &client_address_length);
+		if (clifd == -1)
+		{
+			syslog(LOG_ERR, "Accepting connection failed\n");
+			printf("accept connection failed\n");
+			exit(EXIT_FAILURE);
+		}
+		else
+		{
+			syslog(LOG_INFO, "accept succeeded! Accepted connection from: %s\n", inet_ntoa(client_address.sin_addr));
+			printf("accept connection success\n");
+		}
+
+		//thread parameters
+		data_node = (slist_data_t *) malloc(sizeof(slist_data_t));
+		SLIST_INSERT_HEAD(&head,data_node,entries);
+		data_node->thread_params.clifd           = clifd;
+		data_node->thread_params.thread_complete = false;
+		data_node->thread_params.mutex           = &mutex_lock;
+
+		pthread_create(&(data_node->thread_params.thread_id), 	//Thread ID
+					   NULL,							        //Thread attribute
+					   thread_handler,				            //Thread Handler
+					   &data_node->thread_params );              //Thread parameters
+
+		printf("Threads Created\n");
+
+		SLIST_FOREACH(data_node,&head,entries)
+		{
+			pthread_join(data_node->thread_params.thread_id,NULL);
+			data_node = SLIST_FIRST(&head);
+			SLIST_REMOVE_HEAD(&head, entries);
+			free(data_node);
+		}
+		printf("Threads Excited\n");
+		printf("Closed connection from %s\n",inet_ntoa(client_address.sin_addr));
+	}
 		//close file and clear buffer
 		close(filefd);
-		free(cli_buff);
+		close(sockfd);
 		syslog(LOG_INFO, "Closed connection with %s\n", inet_ntoa(client_address.sin_addr));
+}
+
+
+/*
+ *Function Name: static void timer_handler(int signal)
+ *Description: Signal handler for timer
+ *return: void
+ */
+static void timer_handler(int signal)
+{
+time_t Tmr;
+struct tm *t_ptr;
+int t_length = 0;
+char t_str[200];
+int fd, data_write;
+
+Tmr = time(NULL);
+
+t_ptr = localtime(&Tmr);
+	if(t_ptr == NULL) {
+		syslog(LOG_ERR, "localtime failed\n");
+		perror("Local timer error!");
+		exit(EXIT_FAILURE);
+	}
+
+t_length = strftime(t_str, sizeof(t_str), "timestamp: %m/%d/%Y - %k:%M:%S\n", t_ptr);
+	if (t_length == RESET) {
+		syslog(LOG_ERR, "strftime failed\n");
+		perror("strftime timer error!");
+		exit(EXIT_FAILURE);
 	}
+	
+	printf("%s\n",t_str);
+
+	//writing the time to the file
+	fd = open(FILE_PATH, O_APPEND | O_WRONLY);
+	if(fd == -1) {
+		syslog(LOG_ERR, "File opening failed\n");
+		exit(EXIT_FAILURE);
+	}
+	
+	status = pthread_mutex_lock(&mutex_lock);
+	if (status)
+	{
+		syslog(LOG_ERR, "Mutex lock error %d\n", status);
+		exit(EXIT_FAILURE);
+	}
+	
+	data_write = write(fd, t_str, t_length);
+	if (data_write == -1) {
+		syslog(LOG_ERR, "Fail to write on file\n");
+		exit(EXIT_FAILURE);
```

Comment:
@varun2506 what will happen to mutex lock on this code flow of write failure case?